src/coap_message.c

```diff
@@ -155,23 +155,26 @@ CoAP_Message_t* _rom CoAP_CreateMessage(CoAP_MessageType_t Type,
 		uint16_t PayloadInitialContentLength,
 		uint16_t PayloadMaxSize,
 		CoAP_Token_t Token) {
-	CoAP_Message_t* pMsg = (CoAP_Message_t*) coap_mem_get0(sizeof(CoAP_Message_t) + PayloadMaxSize); //malloc space
-	if (pMsg == NULL)
-		return NULL;
+
+	CoAP_Message_t *pMsg;
 
 	//safety checks
 	if (PayloadInitialContentLength > PayloadMaxSize) {
 		ERROR("Initial Content bigger than field size!");
 		return NULL;
 	}
 
+	pMsg = (CoAP_Message_t*) coap_mem_get0(sizeof(CoAP_Message_t) + PayloadMaxSize); //malloc space
+
+	if (pMsg == NULL)
```

Comment:
Can you please add braces around the body of the if-statement.
And add some error logging. e.g. `ERROR("OutOfMemory: Failed to allocate CoAP_Message_t");`